### PR TITLE
feat(evm): fetch correct amount and receiver address for ERC20 tokens

### PR DESCRIPTION
**Pull request by antbern**
## Motivation and Context

We were only getting the amounts of the underlying transaction (where the receiver is always the smart contract address and value `0`). Instead, with this PR we are now decoding the call parameters and getting the receiver and amount from there for ERC20 impl of `WalletUser`. Also, we were not exposing the receiver address anywhere so I added that 👍 

## Checklist

Please ensure that your PR meets the following requirements:

### General

- [ ] Code follows project style and guidelines.
- [ ] No redundant or duplicate code.
- [ ] No added new dependencies without clear justification and approval.

### Documentation

- [ ] Added/updated relevant documentation (e.g., README, inline comments).
- [ ] Added a CHANGELOG entry for major changes.

### Testing

- [ ] Added relevant test cases, leveraging tools like `rstest` or similar for reusable tests.
- [ ] Tests pass locally.

### Build & CI/CD

- [ ] Confirmed no CI/CD pipeline issues.
- [ ] Updated build scripts where necessary.

## Comments to Reviewer

Please provide any comments or points of consideration for the reviewers, such as specific areas of focus, potential edge cases, or context that would aid the review process.

### Reviewer Responsibilities

- [ ] Code adheres to style and guidelines.
- [ ] All tests are appropriately covered and they pass.
- [ ] Consider performance and/or security impacts.
- [ ] Constructive feedback is provided.

### CODE REVIEW

File: sdk/examples/13_send_amount.rs [hunk 1]
```diff
@@ -47,4 +47,7 @@ async fn main() {
         .unwrap();
 
     println!("Success with transaction id: {tx_id}");
+
+    let details = sdk.get_wallet_tx(&user.pin, &tx_id).await.unwrap();
+    println!("Details:\n{:#?}", details);
```
Comment (kristifidani) [inline sdk/examples/13_send_amount.rs:52]: Do we want to print these details also for the binding's examples? 😄

**Review by kristifidani — APPROVED** [review]
LGTM 💯

**Review by kristifidani — APPROVED** [review]

**Review by kristifidani — APPROVED** [review]

**Review by hauju — APPROVED** [review]
